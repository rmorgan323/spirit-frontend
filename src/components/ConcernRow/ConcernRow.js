import React, { Component } from 'react';
import './ConcernRow.css';



class ConcernRow extends Component {
  constructor(props) {
    super(props);

    this.state = {
      hover: false
    }
  } 

  selectEnter = () => {
    this.setState({hover: true})

    let activeDomains = Object.keys(this.props).reduce((accum, prop) => {
      if (prop.length === 2 && this.props[prop]) {
        const y = [...prop].pop();
        accum.push(y)
      }
      return accum;
    },[])

    const samDots = [].slice.call(document.querySelectorAll('.sam-dots'))
    samDots.forEach((dot, index) => {
      if (activeDomains.includes((index + 1).toString())) {
        document.querySelector(`.sam-dot-${index + 1}`).classList.add('sam-hover')
      }
    })
    
  }

  selectLeave = () => {
    this.setState({hover: false})

    const samDots = [].slice.call(document.querySelectorAll('.sam-dots'))
    samDots.forEach(dot => {
      dot.classList.remove('sam-hover')
    })
  }

  render() {
    return (
      <div className="ConcernRow">
        <p className="concern-left">{this.props.concern}</p>
        <p>
          <span className={ this.props.d1 === false ? "domains" : (this.state.hover === false ? "domains domains-true" : "domains domains-true domains-hover") }>1</span>
          <span className={ this.props.d2 === false ? "domains" : (this.state.hover === false ? "domains domains-true" : "domains domains-true domains-hover") }>2</span>
          <span className={ this.props.d3 === false ? "domains" : (this.state.hover === false ? "domains domains-true" : "domains domains-true domains-hover") }>3</span>
          <span className={ this.props.d4 === false ? "domains" : (this.state.hover === false ? "domains domains-true" : "domains domains-true domains-hover") }>4</span>
          <span className={ this.props.d5 === false ? "domains" : (this.state.hover === false ? "domains domains-true" : "domains domains-true domains-hover") }>5</span>
          <span className={ this.props.d6 === false ? "domains" : (this.state.hover === false ? "domains domains-true" : "domains domains-true domains-hover") }>6</span>
        </p>
        <p className="concern-notes">{this.props.notes}</p>
        <button onMouseEnter={() => this.selectEnter()} onMouseOut={() => this.selectLeave()} className="select-concern-button">SELECT</button>
      </div>
    )
  }
}

export default ConcernRow;