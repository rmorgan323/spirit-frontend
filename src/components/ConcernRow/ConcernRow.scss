@import '../../variables.scss';

.ConcernRow {

  display: flex;

  .domains {
    opacity: 0;
    cursor: unset;
    @include std-transition;
  }

  .domains-true {
    opacity: 1;
  }

  .domains-hover {
    border: 5px solid $purple5;
  }

  .concern-left, .concern-notes {
    width: 304px;
    text-align: left;
    background-color: $grey1;
    padding: 13px 8px;
    margin: 0 0 28px;
  }

  .concern-left {
    margin-right: 29px;
  }

  .concern-notes {
    margin-left: 28px;
  }

  .select-concern-button {
    width: 130px;
    height: 44px;
    background-color: $grey3;
    color: $grey1;
    margin-left: 20px;
    border: none;
    cursor: pointer;
  }

  .select-concern-button:hover {
    background-color: $grey4;
  }
}