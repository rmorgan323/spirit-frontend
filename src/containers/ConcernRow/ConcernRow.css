/******  GLOBAL CSS VARIABLES  ******/
/***  COLORS  ***/
/***  FONTS  ***/
* {
  font-size: 16px;
  font-family: "Lato", sans-serif;
  outline: none;
  color: #6d6e71; }

h1, h2, h3, h4, h5, h6 {
  font-family: "Josefin Sans", sans-serif;
  line-height: 1.3; }

h1 {
  font-size: 44px; }

h2 {
  font-size: 30px; }

h3 {
  font-size: 24px; }

h4 {
  font-size: 18px; }

h5 {
  font-size: 14px; }

h6 {
  font-size: 10px; }

p {
  font-size: 14px;
  line-height: 1.3;
  font-weight: 900; }

::placeholder {
  opacity: .5;
  text-align: center; }

/***  MIXINS  ***/
.ConcernRow {
  display: flex; }
  .ConcernRow .domains {
    opacity: 0;
    cursor: unset;
    transition: all .2s ease-in-out; }
  .ConcernRow .domains-true {
    opacity: 1; }
  .ConcernRow .domains-hover {
    border: 5px solid #7F60A4; }
  .ConcernRow .concern-left, .ConcernRow .concern-notes {
    width: 304px;
    text-align: left;
    background-color: #f1f2f2;
    padding: 13px 8px;
    margin: 0 0 28px; }
  .ConcernRow .concern-left {
    margin-right: 29px; }
  .ConcernRow .concern-notes {
    margin-left: 28px; }
  .ConcernRow .select-concern-button {
    width: 130px;
    height: 44px;
    background-color: #a7a9ac;
    color: #f1f2f2;
    margin-left: 20px;
    border: none;
    cursor: pointer; }
  .ConcernRow .select-concern-button:hover {
    background-color: #808285; }
