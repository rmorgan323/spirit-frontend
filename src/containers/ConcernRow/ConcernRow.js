import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import * as actions from '../../actions/index';
import './ConcernRow.css';

export class ConcernRow extends Component {
  state = {
    hover: false
  };

  selectEnter = () => {
    this.setState({ hover: true });

    let activeDomains = Object.keys(this.props).reduce((accum, prop) => {
      if (prop.length === 2 && this.props[prop]) {
        const activeDomain = [...prop].pop();
        accum.push(activeDomain);
      }
      return accum;
    }, []);

    const samDots = [].slice.call(document.querySelectorAll('.sam-dots'));
    samDots.forEach((dot, index) => {
      if (activeDomains.includes((index + 1).toString())) {
        document
          .querySelector(`.sam-dot-${index + 1}`)
          .classList.add('sam-hover');
      }
    });
  };

  selectLeave = () => {
    this.setState({ hover: false });

    const samDots = [].slice.call(document.querySelectorAll('.sam-dots'));
    samDots.forEach(dot => {
      dot.classList.remove('sam-hover');
    });
  };

  determineDomainClass = (domain, hover) => {
    if (!domain) {
      return 'domains';
    } else if (!hover) {
      return 'domains domains-true';
    } else {
      return 'domains domains-true domains-hover';
    }
  };

  render() {
    const { id, concern, d1, d2, d3, d4, d5, d6, notes } = this.props;
    const { hover } = this.state;

    return (
      <div className="ConcernRow">
        <p className="concern-left">{concern}</p>

        <p>
          <span className={this.determineDomainClass(d1, hover)}>1</span>
          <span className={this.determineDomainClass(d2, hover)}>2</span>
          <span className={this.determineDomainClass(d3, hover)}>3</span>
          <span className={this.determineDomainClass(d4, hover)}>4</span>
          <span className={this.determineDomainClass(d5, hover)}>5</span>
          <span className={this.determineDomainClass(d6, hover)}>6</span>
        </p>

        <p className="concern-notes">{notes}</p>

        <Link to={`/spirit/concerns/${id}/sessions`}>
          <button
            className="select-concern-button"
            onClick={() => this.props.getConcern(id)}
            onMouseEnter={() => this.selectEnter()}
            onMouseOut={() => this.selectLeave()}
          >
            Start Process
          </button>
        </Link>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  getConcern: concernId => dispatch(actions.getConcern(concernId))
});

export default connect(
  null,
  mapDispatchToProps
)(ConcernRow);

ConcernRow.propTypes = {
  id: PropTypes.number,
  concern: PropTypes.string,
  d1: PropTypes.bool,
  d2: PropTypes.bool,
  d3: PropTypes.bool,
  d4: PropTypes.bool,
  d5: PropTypes.bool,
  d6: PropTypes.bool,
  notes: PropTypes.string,
  getConcern: PropTypes.func
};
