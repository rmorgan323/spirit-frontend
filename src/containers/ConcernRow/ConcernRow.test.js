/*eslint-disable max-len*/

import React from 'react';
import { ConcernRow, mapDispatchToProps } from './ConcernRow';
import { shallow } from 'enzyme';

describe('ConcernRow tests', () => {
  let renderedConcernRow;
  let mockGetConcern;

  beforeEach(() => {
    mockGetConcern = jest.fn();

    renderedConcernRow = shallow(<ConcernRow getConcern={mockGetConcern} />);
  });

  it('Should match the snapshot', () => {
    expect(renderedConcernRow).toMatchSnapshot();
  });

  it('should set state when selectEnter is called', () => {
    expect(renderedConcernRow.state('hover')).toEqual(false);
    renderedConcernRow.instance().selectEnter();

    expect(renderedConcernRow.state('hover')).toEqual(true);
  });

  it('should set state when selectLeave is called', () => {
    expect(renderedConcernRow.state('hover')).toEqual(false);
    renderedConcernRow.instance().selectEnter();

    expect(renderedConcernRow.state('hover')).toEqual(true);
    renderedConcernRow.instance().selectLeave();

    expect(renderedConcernRow.state('hover')).toEqual(false);
  });

  it('determineDomainClass should return "domains" if domain value is false', () => {
    expect(renderedConcernRow.instance().determineDomainClass(false, false)).toEqual('domains');
  });

  it('determineDomainClass should return "domains domains-true" if domain value is true', () => {
    expect(renderedConcernRow.instance()
      .determineDomainClass(true, false))
      .toEqual('domains domains-true');
  });

  it('determineDomainClass should return "domains domains-true domains-hover" if domains and hover values are true', () => {
    expect(renderedConcernRow.instance()
      .determineDomainClass(true, true))
      .toEqual('domains domains-true domains-hover');
  });
});

describe('mapDispatchToProps tests', () => {
  it('Should call dispatch when getConcern is called', () => {
    const mockDispatch = jest.fn();
    const mockParameters = 5;
    const result = mapDispatchToProps(mockDispatch);

    result.getConcern(mockParameters);
    expect(mockDispatch).toHaveBeenCalled();
  });
});
