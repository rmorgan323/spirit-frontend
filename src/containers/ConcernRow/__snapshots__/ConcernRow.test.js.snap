// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConcernRow tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConcernRow
    getConcern={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="concern-left"
        />,
        <p>
          <span
            className="domains"
          >
            1
          </span>
          <span
            className="domains"
          >
            2
          </span>
          <span
            className="domains"
          >
            3
          </span>
          <span
            className="domains"
          >
            4
          </span>
          <span
            className="domains"
          >
            5
          </span>
          <span
            className="domains"
          >
            6
          </span>
        </p>,
        <p
          className="concern-notes"
        />,
        <Link
          replace={false}
          to="/spirit/concerns/undefined/sessions"
        >
          <button
            className="select-concern-button"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseOut={[Function]}
          >
            SELECT
          </button>
        </Link>,
      ],
      "className": "ConcernRow",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "concern-left",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="domains"
            >
              1
            </span>,
            <span
              className="domains"
            >
              2
            </span>,
            <span
              className="domains"
            >
              3
            </span>,
            <span
              className="domains"
            >
              4
            </span>,
            <span
              className="domains"
            >
              5
            </span>,
            <span
              className="domains"
            >
              6
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1",
              "className": "domains",
            },
            "ref": null,
            "rendered": "1",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2",
              "className": "domains",
            },
            "ref": null,
            "rendered": "2",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "3",
              "className": "domains",
            },
            "ref": null,
            "rendered": "3",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "4",
              "className": "domains",
            },
            "ref": null,
            "rendered": "4",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "5",
              "className": "domains",
            },
            "ref": null,
            "rendered": "5",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "6",
              "className": "domains",
            },
            "ref": null,
            "rendered": "6",
            "type": "span",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "concern-notes",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="select-concern-button"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseOut={[Function]}
          >
            SELECT
          </button>,
          "replace": false,
          "to": "/spirit/concerns/undefined/sessions",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SELECT",
            "className": "select-concern-button",
            "onClick": [Function],
            "onMouseEnter": [Function],
            "onMouseOut": [Function],
          },
          "ref": null,
          "rendered": "SELECT",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="concern-left"
          />,
          <p>
            <span
              className="domains"
            >
              1
            </span>
            <span
              className="domains"
            >
              2
            </span>
            <span
              className="domains"
            >
              3
            </span>
            <span
              className="domains"
            >
              4
            </span>
            <span
              className="domains"
            >
              5
            </span>
            <span
              className="domains"
            >
              6
            </span>
          </p>,
          <p
            className="concern-notes"
          />,
          <Link
            replace={false}
            to="/spirit/concerns/undefined/sessions"
          >
            <button
              className="select-concern-button"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseOut={[Function]}
            >
              SELECT
            </button>
          </Link>,
        ],
        "className": "ConcernRow",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "concern-left",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="domains"
              >
                1
              </span>,
              <span
                className="domains"
              >
                2
              </span>,
              <span
                className="domains"
              >
                3
              </span>,
              <span
                className="domains"
              >
                4
              </span>,
              <span
                className="domains"
              >
                5
              </span>,
              <span
                className="domains"
              >
                6
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1",
                "className": "domains",
              },
              "ref": null,
              "rendered": "1",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2",
                "className": "domains",
              },
              "ref": null,
              "rendered": "2",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "3",
                "className": "domains",
              },
              "ref": null,
              "rendered": "3",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "4",
                "className": "domains",
              },
              "ref": null,
              "rendered": "4",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "5",
                "className": "domains",
              },
              "ref": null,
              "rendered": "5",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "6",
                "className": "domains",
              },
              "ref": null,
              "rendered": "6",
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "concern-notes",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="select-concern-button"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseOut={[Function]}
            >
              SELECT
            </button>,
            "replace": false,
            "to": "/spirit/concerns/undefined/sessions",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SELECT",
              "className": "select-concern-button",
              "onClick": [Function],
              "onMouseEnter": [Function],
              "onMouseOut": [Function],
            },
            "ref": null,
            "rendered": "SELECT",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
